src/java/org/apache/cassandra/security/DefaultCryptoProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.security;
+
+import java.util.Map;
+import javax.crypto.Cipher;
+
+import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
+
+/**
+ * Default crypto provider tries to install AmazonCorrettoCryptoProvider.
+ * <p>
+ * The implementation falls back to in-built crypto provider in JRE if the installation
+ * is not successful.
+ */
+public class DefaultCryptoProvider extends AbstractCryptoProvider
+{
+    public DefaultCryptoProvider(Map<String, String> args)
+    {
+        super(args);
+    }
+
+    @Override
+    public String getProviderName()
+    {
+        return AmazonCorrettoCryptoProvider.PROVIDER_NAME;
+    }
+
+    @Override
+    public String getProviderClassAsString()
+    {
+        return "com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider";
```

Comment:
The reason it is `String` is that if it is not on the class path, then it would throw what ... `ClassNotFoundException`? So I made it String on purpose and I am using `Class.forName` (which can be changed to FBUtilities as you suggested)

Reply:
I saw that the class already imports the `AmazonCorrettoCryptoProvider`, that's why suggested this change. If you want Strings then both `getProviderName` and `getProviderClassAsString` should be simple strings :-)

Reply:
+1 for use of FBUtilities